src/query/service/src/pipelines/processors/transforms/hash_join/probe_join/right_mark.rs

```diff
@@ -44,8 +44,7 @@ impl JoinHashTable {
     {
         let valids = &probe_state.valids;
         let has_null = *self.hash_join_desc.marker_join_desc.has_null.read();
-        let mut markers = Self::init_markers(input.columns(), input.num_rows());
```

Comment:
Bug in right mark join: we should use `probe_keys` to init markers, not probe input columns. cc @youngsofun